src/Users/UserStatistics.php

```diff
@@ -0,0 +1,341 @@
+<?php
+
+namespace Scriptotek\Alma\Users;
+
+use Scriptotek\Alma\Client;
+use Scriptotek\Alma\Model\Model;
+
+/**
+ * Iterable collection of statistics resources.
+ */
+class UserStatistics extends Model
+{
+    /**
+     * Get the user's statistics.
+     *
+     * @return Array of Statistics.
+     */
+    public function getStatistics()
+    {
+        $stats = array();
+        foreach ($this->data->user_statistic as $statistic) {
+            $stats[] = $statistic;
+        }
+        return $stats;
+    }
+
+    /**
+     * Get the user's statistics.
+     * 
+     * @return array of statistics.
+     */
+    public function get()
+    {
+        $stats = array();
+        foreach ($this->data->user_statistic as $statistic) {
+            $stats[] = $statistic;
+        }
+        return $stats;
+    }
+
+    /**
+    * Get a statistic.
+    *
+    * @param $typeCode code of the category type.
+    * @param $categoryCode code of the statistical category.
+    *
+    * @return Statistic.
+    */
+    public function getStatistic($typeCode,$categoryCode)
+    {
+        $stats = array();
+        foreach ($this->data->user_statistic as $statistic) {
+            if (($statistic->category_type->value == $typeCode) && 
+                ($statistic->statistic_category->value == $categoryCode)) {
+                $stats[] = $statistic;
+            }
+        }
+        return $stats;
+    }
+
+    /**
+    * Add a user statistic with no regard to existing Types and Categories.
+    *
+    * @param string $typeCode code of the category type.
+    * @param string $typeDesc description of the category type.
+    * @param string $categoryCode code of the statistical category.
+    * @param string $categoryDesc description of the statistical category.
+    * @param string $note free text description of the statistic.
+    * @param string $segment_type either "Internal" or "External"
+    */
+    public function addStatisticRaw($typeCode,$typeDesc,$categoryCode,$categoryDesc,$segment_type = 'External' ,$note = '')
```

Comment:
Suggested refactor:
`addStatistic()` adds the values unchecked (in line with the fact that such invalid data can exist in Alma

But add a `Statistics->Statistic->validate()` method, and `Statistics->validate()` method to confirm the statistics against actual codetables.

Warning: this has substantial implications for the Lazy Load approach, and for caching of codetables.